Tests/MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -151,6 +151,38 @@ class RouteControllerTests: TestCase {
 
         waitForExpectations(timeout: 1, handler: nil)
     }
+    
+    func testAlternativeRoutesReported() {
+        let routeOptions = RouteOptions(coordinates: [.init(latitude: 37.33243586131637,
+                                                            longitude: -122.03140541047281),
+                                                      .init(latitude: 37.33318065375225,
+                                                            longitude: -122.03148874952787)],
+                                        profileIdentifier: .automobileAvoidingTraffic)
+        routeOptions.shapeFormat = .geoJSON
+        let routeResponse = Fixture.routeResponse(from: "routeResponseWithAlternatives",
+                                                  options: routeOptions)
+        
+        let alternativesExpectation = XCTestExpectation(description: "Alternative route should be reported")
+        alternativesExpectation.assertForOverFulfill = true
+        
+        let observer = AlternativeRoutesObserver()
+        observer.onDidReportAlternatives = { newAlternatives, removedAlternatives in
+            XCTAssertTrue(removedAlternatives.isEmpty)
+            if newAlternatives.count == 1 {
+                alternativesExpectation.fulfill()
```

Comment:
Should we add an assert for `newAlternatives` size and fulfill the expectation in any case?

Reply:
Navigator is designed that way, so that when it is initialized it instantly reports `0 alternatives`, even before a route was set.